airflow/providers/smtp/hooks/smtp.py

```diff
@@ -109,6 +109,24 @@ def _build_client(self) -> smtplib.SMTP_SSL | smtplib.SMTP:
             smtp_kwargs["port"] = self.port
         smtp_kwargs["timeout"] = self.timeout
 
+        if self.use_ssl:
+            from airflow.configuration import conf
+
+            ssl_context_string = conf.get("smtp_provider", "SSL_CONTEXT", fallback=None)
+            if ssl_context_string is None:
+                ssl_context_string = conf.get("email", "SSL_CONTEXT", fallback=None)
+            if ssl_context_string is None:
+                ssl_context_string = "default"
```

Comment:
```suggestion
            if ssl_context_string is None:
                ssl_context_string = conf.get("email", "SSL_CONTEXT", fallback="default")
```

At this point can we fallback to `default`? or we would like to have an explicit check later? or the fallback has some disadvantages? just curios :)

Reply:
We do fallback to default. The complexity is that we really want to have an "smtp-provider" default separate from the "email" one. This is why we do not have "built-in" default for the "smtp_provider" one. If we have  a default there, we cannot really use "email" as a fallback if it is set in Airflow.